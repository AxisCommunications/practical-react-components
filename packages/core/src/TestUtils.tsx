import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { PracticalProvider } from './Practical'

/*
  Only used for Jest Snapshot Testing
*/

export const TP = ({
  children,
}: {
  readonly children: React.ReactElement
}): JSX.Element => <PracticalProvider>{children}</PracticalProvider>

/**
 * Mock reference to practical-root
 */
const createNodeMock = (element: React.ReactElement) => {
  if (element.props.id !== 'practical-root') {
    return null
  }

  return {}
}

export const TestRender = (jsx: JSX.Element) => {
  return renderer.create(<TP>{jsx}</TP>, { createNodeMock }).toJSON()
}

export const TestText = (): JSX.Element => <span>Test</span>

/* eslint-disable-next-line react/display-name */
export const TestTextWithForwardRef = React.forwardRef<HTMLElement>(
  ({ ...props }, ref) => (
    <span {...props} ref={ref}>
      Test
    </span>
  )
)
