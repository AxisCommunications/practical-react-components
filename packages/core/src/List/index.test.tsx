import React from 'react'
import 'jest-styled-components'

import { List, ListItem } from '.'
import { TestRender, TestText } from '../TestUtils'

describe('Lists', () => {
  test('List', () => {
    const tree1 = TestRender(
      <List>
        <ListItem>
          <TestText />
        </ListItem>
        <ListItem>
          <TestText />
        </ListItem>
        <ListItem>
          <TestText />
        </ListItem>
      </List>
    )
    expect(tree1).toMatchSnapshot()
  })
})
