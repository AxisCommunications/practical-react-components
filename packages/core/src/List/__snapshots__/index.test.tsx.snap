// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lists List 1`] = `
.c2 {
  margin: 0;
  padding: 0;
  overflow-y: auto;
  color: rgb(41,41,41);
}

.c3 {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  height: 40px;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
}

.c3:hover {
  background-color: rgb(245,245,245);
}

.c5 {
  position: fixed;
  top: 0;
  width: 360px;
  border-radius: 4px;
  margin: 8px 16px;
  right: 0;
}

.c0 {
  color: rgb(41,41,41);
}

.c1 {
  position: absolute;
  z-index: 0;
}

.c4 {
  position: absolute;
  z-index: 1;
}

<div
  className="c0"
  id="practical-root"
>
  <div>
    <div
      className="c1"
      id="layer-1"
    >
      <ul
        className="c2"
      >
        <li
          className="c3"
        >
          <span>
            Test
          </span>
        </li>
        <li
          className="c3"
        >
          <span>
            Test
          </span>
        </li>
        <li
          className="c3"
        >
          <span>
            Test
          </span>
        </li>
      </ul>
    </div>
  </div>
  <div>
    <div
      className="c4"
      id="layer-2"
    >
      <div
        className="c5"
      />
    </div>
  </div>
</div>
`;
